packages/0x.js/src/order_watcher/order_state_watcher.ts

```diff
@@ -84,6 +90,8 @@ export class OrderStateWatcher {
         assert.isValidSignature(orderHash, signedOrder.ecSignature, signedOrder.maker);
         this._orderByOrderHash[orderHash] = signedOrder;
         this.addToDependentOrderHashes(signedOrder, orderHash);
+        // We don't remove orders from expirationWatcher because heap removal is linear. We just skip it later
```

Comment:
Won't the heap grow into a huge memory leak if tons of orders with very length expirations are never removed from it?

Reply:
Will fix in a next PR